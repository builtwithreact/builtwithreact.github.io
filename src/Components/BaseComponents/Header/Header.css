.HeaderContainer {
  background-color: #1c1d21;
  height: 60px; }
  .HeaderContainer.HeaderContainer__Sticky {
    position: fixed;
    width: 100%; }
  .HeaderContainer .Header {
    display: flex;
    justify-content: space-between;
    align-items: center; }
  .HeaderContainer .Logo {
    color: #fff;
    display: flex;
    text-decoration: none;
    font-size: 24px;
    font-weight: bold;
    align-items: center;
    font-family: Courgette, sans-serif; }
    .HeaderContainer .Logo img {
      width: 70px;
      height: 60px; }

.Navigation ul {
  list-style: none;
  margin: 0;
  padding: 0;
  flex-grow: 1;
  margin-left: auto;
  display: flex;
  align-items: center; }
  .Navigation ul .Navigation-Link {
    color: #fff;
    text-decoration: none;
    padding: 20px 10px;
    font-family: Courgette, sans-serif;
    font-size: 18px;
    line-height: 1; }
    .Navigation ul .Navigation-Link:hover {
      color: #4caf4f; }
