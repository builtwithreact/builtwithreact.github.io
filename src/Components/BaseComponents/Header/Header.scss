.HeaderContainer {
  background-color: #1c1d21;
  height: 60px;

  .Header {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .Logo {
    color: #4caf4f;
    display: flex;
    text-decoration: none;
    font-size: 20px;
    font-weight: bold;
    align-items: center;
    img {
      width: 70px;  
      height: 60px;
    }
  }
}

.Navigation {
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
    flex-grow: 1;
    margin-left: auto;
    display: flex;
    align-items: center;

    .Navigation-Link {
      color: #fff;
      text-decoration : none;
      padding: 20px 10px;
      &:hover {
        color: #4caf4f;
      }
    }
  }
}
