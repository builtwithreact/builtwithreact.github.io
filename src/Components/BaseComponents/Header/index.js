import React, { Component } from 'react';
import { Link, NavLink, Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import logo from './../../../Assets/images/logo.svg';

import './Header.css';

class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            stickyHeader: false
        }
    }

    componentDidMount() {
        document.addEventListener('scroll', (e) => {
            if (window.scrollY > 60) {
                this.setState({ stickyHeader: true });
            } else {
                this.setState({ stickyHeader: false });
            }
        })
    }
    render() {
        const navitems = [
            { "title": "Web", "url": "ReactWeb" },
            { "title": "React Native", "url": "ReactNative" },
            { "title": "Blogs", "url": "ReactBlogs" },
            { "title": "Tools", "url": "ReactTools" }
        ]

        return (
            <div className={`HeaderContainer ${this.state.stickyHeader ? 'HeaderContainer__Sticky' : ''}`}>
                <div className="Header container">
                    <Link to="/" className="Logo">
                        <img src={logo} alt="Built With React" />
                        <span>Built with React</span>
                    </Link>
                    <nav className="Navigation">
                        <ul>
                            {
                                navitems.map((item) => (
                                    <li className="Navigation-Item" key={`nav-${item.title.replace(" ", "-")}`}>
                                        <NavLink className="Navigation-Link"
                                            activeStyle={{ color: "#4caf4f", borderBottom: "2px #4caf4f solid" }}
                                            to={`/${item.url}`}>
                                            {item.title}
                                        </NavLink>
                                    </li>
                                ))
                            }
                        </ul>
                    </nav>
                </div>
            </div>
        );
    }
}

Header.propTypes = {

};

export default Header;