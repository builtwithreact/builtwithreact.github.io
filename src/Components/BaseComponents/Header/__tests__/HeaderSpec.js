import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import { StaticRouter } from 'react-router-dom'
import Header from './../index';

describe('Header', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <Header />
            </StaticRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
