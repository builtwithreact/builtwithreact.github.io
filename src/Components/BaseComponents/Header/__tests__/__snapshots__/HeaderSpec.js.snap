// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="HeaderContainer "
>
  <div
    className="Header container"
  >
    <a
      className="Logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Built With React"
        src="logo.svg"
      />
      <span>
        Built with React
      </span>
    </a>
    <nav
      className="Navigation"
    >
      <ul>
        <li
          className="Navigation-Item"
        >
          <a
            aria-current={null}
            className="Navigation-Link"
            href="/ReactWeb"
            onClick={[Function]}
            style={undefined}
          >
            Web
          </a>
        </li>
        <li
          className="Navigation-Item"
        >
          <a
            aria-current={null}
            className="Navigation-Link"
            href="/ReactNative"
            onClick={[Function]}
            style={undefined}
          >
            React Native
          </a>
        </li>
        <li
          className="Navigation-Item"
        >
          <a
            aria-current={null}
            className="Navigation-Link"
            href="/ReactBlogs"
            onClick={[Function]}
            style={undefined}
          >
            Blogs
          </a>
        </li>
        <li
          className="Navigation-Item"
        >
          <a
            aria-current={null}
            className="Navigation-Link"
            href="/ReactTools"
            onClick={[Function]}
            style={undefined}
          >
            Tools
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
